pkg/controllers/collaset/synccontrol/sync_control.go

```diff
@@ -285,6 +281,23 @@ func (r *RealSyncControl) SyncPods(
 		}
 	}
 
+	// 5. Reclaim podToDelete if necessary
+	if len(toDeletePodNames) > 0 {
+		var newPodToDelete []string
+		for _, podName := range instance.Spec.ScaleStrategy.PodToDelete {
+			if !toDeletePodNames.Has(podName) {
+				newPodToDelete = append(newPodToDelete, podName)
+			}
+		}
+		instance.Spec.ScaleStrategy.PodToDelete = newPodToDelete
+		// update cls.spec.scaleStrategy.podToDelete
+		if err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+			return r.client.Update(ctx, instance)
+		}); err != nil {
+			return false, nil, ownedIDs, fmt.Errorf("fail to update collaset when reclaiming podToDelete: %s", err)
+		}
+	}
```

Comment:
I don't think it's reasonable to update the `spec` in controller reconciliation

Reply:
enhanced by adding a featureGate ReclaimPodToDelete（default true）:601a0e328f2b543fe43f83157eab9725dc6b3136